site/src/layouts/DetailComponent/DetailComponent.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC, useCallback, useMemo } from "react";
+import { useRouter } from "next/navigation";
+import {
+  TableOfContents,
+  Sidebar,
+} from "@jpmorganchase/mosaic-site-components";
+import { useRoute, useStore, SiteState } from "@jpmorganchase/mosaic-store";
+import { TabPanel, Tabs } from "@salt-ds/lab";
+import { LayoutProps } from "../types/index";
+import { DetailBase } from "../DetailBase";
+
+const tabs = [
+  { id: 0, name: "examples", label: "Examples" },
+  { id: 1, name: "usage", label: "How to use" },
+  { id: 2, name: "accessibility", label: "Accessibility" },
+];
+
+type CustomSiteState = SiteState & { data?: Record<string, any> };
+
+export const DetailComponent: FC<LayoutProps> = ({ children }) => {
+  const { push } = useRouter();
+  const { route } = useRoute();
+
+  const newRoute = route?.substring(0, route.lastIndexOf("/") + 1);
+
+  const useData = useStore((state: CustomSiteState) => state.data);
+
+  const { description } = useData || {};
+
+  const updateRouteWhenTabChanges = useCallback((index: number) => {
+    const currentTab = tabs.find(({ id }) => index === id);
+
+    currentTab
+      ? push(`${newRoute}${currentTab.name}`)
+      : push(`${newRoute}${tabs[0].name}`);
+  }, []);
+
+  const updateTabWhenRouteChanges = useMemo(() => {
+    const currentTab = tabs.find(({ name }) => route?.includes(name));
+
+    const defaultTabAndRoute = () => {
```

Comment:
Do you know if we need to handle this at all? Either way, it will need to be in a useEffect

Reply:
Yeah we need to handle the scenario where the user clicks or navigates to the Button root page (in which case we want to display the first page "examples" and select the first tab). Do you mean only this specific scenario should be in a useEffect?